lib/Coocook/Controller/Dish.pm

```diff
@@ -85,6 +85,26 @@ sub edit : GET HEAD Chained('base') PathPart('') Args(0) RequiresCapability('vie
 
 sub delete : POST Chained('base') PathPart('delete') Args(0) RequiresCapability('edit_project') {
     my ( $self, $c ) = @_;
+    my $dish = $c->stash->{dish};
+    for my $ingredient ( $dish->ingredients->all ) {
+        my $item = $ingredient->item;
+        if ( $item ? not $item->ingredients->exists : 1 ) {
+            $item->delete() if $item;
+            next;
+        }
+        next unless $item;
```

Comment:
Seems like there is more code than needed.

Suppose `$item` is not defined. Then `$item ? not $item->ingredients->exists : 1` will evaluate to `1`, causing the `if`'s body to be executed. Because of our assumption, `$item->delete()` won't fire. So, effectively, the body will execute `next`.
Now, let's look at line 95. If instead, the execution would skip the `if`'s body (while `$item` is still undefined) the `next` statement in `next unless $item` would be executed.

This means, that no matter if we execute the `if` body or not while `$item` is undefined, a `next` will be fired.
Because of that, we can simplify the lines 91 to 95 to:
```
next unless $item;
if ( not $item->ingredients->exists ) {
    $item->delete();
    next;
}
```

Reply:
Yes, that seems to be a simplification here.

I suggest writing `$item or next;` instead of `next unless $item` to keep the order of execution. Some Perl programmers dislike trailing `if`/`unless` because it reverts the order of execution. I have no strong opinion on this one.

Reply:
I'm no expert on this since I don't really know how to program in Perl. This suggestion was only based on the knowledge which I gained by reading the code and trying to understand the meaning of the syntax. So I leave the actual change to the experts.